llvm/lib/CodeGen/SelectionDAG/SelectionDAG.cpp

```diff
@@ -9854,6 +9854,27 @@ SDValue SelectionDAG::getNode(unsigned Opcode, const SDLoc &DL, SDVTList VTList,
       SDValue ZeroOverFlow = getConstant(0, DL, VTList.VTs[1]);
       return getNode(ISD::MERGE_VALUES, DL, VTList, {N1, ZeroOverFlow}, Flags);
     }
+
+    if (VTList.VTs[0].isVector() &&
+        VTList.VTs[0].getVectorElementType() == MVT::i1 &&
+        VTList.VTs[1].getVectorElementType() == MVT::i1) {
```

Comment:
Why limit this to vectors only?

Reply:
Mainly because all other add/sub/sat cases did

Reply:
I think it would make sense to switch all of them to work on scalars as well, but that can be a separate change...